libraries/cyclestreets-view/src/main/java/net/cyclestreets/routing/Route.kt

```diff
@@ -0,0 +1,244 @@
+    }
+
+    @JvmStatic
+    fun RenameRoute(localId: Int, newName: String) {
+        db_.renameRoute(localId, newName)
+    }
+
+    @JvmStatic
+    fun DeleteRoute(localId: Int) {
+        db_.deleteRoute(localId)
+    }
+
+    /////////////////////////////////////////
+    private var plannedRoute_ = NULL_JOURNEY
+    private var waypoints_ = plannedRoute_.waypoints
+    private lateinit var db_: RouteDatabase
+    private lateinit var context_: Context
+    @JvmStatic
+    fun initialise(context: Context) {
+        context_ = context
+        db_ = RouteDatabase(context)
+        if (isLoaded)
+            loadLastJourney()
+        else
+            restoreWaypoints()
+    }
+
+    fun resetJourney() {
+        onNewJourney(null)
+    }
+
+    fun onPause(waypoints: Waypoints) {
+        if (!isLoaded) {
+            waypoints_ = waypoints
+            stashWaypoints()
+        }
+    }
+
+    fun onResume() {
+        Segment.formatter = DistanceFormatter.formatter(CycleStreetsPreferences.units())
+    }
+
+    /////////////////////////////////////
+    fun storedCount(): Int {
+        return db_.routeCount()
+    }
+
+    @JvmStatic
+    fun storedRoutes(): List<RouteSummary> {
+        return db_.savedRoutes()
+    }
+
+    /////////////////////////////////////
+    fun onNewJourney(route: RouteData?): Boolean {
+        try {
+            doOnNewJourney(route)
+            return true
+        } catch (e: Exception) {
+            Log.w(TAG, "Route finding failed", e)
+            Toast.makeText(context_, R.string.route_finding_failed, Toast.LENGTH_LONG).show()
+        }
+        return false
+    }
+
+    private fun doOnNewJourney(route: RouteData?) {
+        if (route == null) {
+            plannedRoute_ = NULL_JOURNEY
+            waypoints_ = NULL_WAYPOINTS
+            listeners_.onReset()
+            clearRouteLoaded()
+            return
+        }
+        plannedRoute_ = loadFromJson(route.json(), route.points(), route.name())
+        db_.saveRoute(plannedRoute_, route.json())
+        waypoints_ = plannedRoute_.waypoints
+        listeners_.onNewJourney(plannedRoute_, waypoints_)
+        setRouteLoaded()
+    }
+
+    fun waypoints(): Waypoints {
+        return waypoints_
+    }
+
+    @JvmStatic
+    fun available(): Boolean {
+        return plannedRoute_ != NULL_JOURNEY
+    }
+
+    @JvmStatic
+    fun journey(): Journey {
+        return plannedRoute_
+    }
+
+    private fun loadLastJourney() {
+        val routeSummaries = storedRoutes()
+        if (!storedRoutes().isEmpty()) {
+            val lastRoute = routeSummaries[0]
+            val route = db_.route(lastRoute.localId())
+            onNewJourney(route)
+        }
+    }
+
+    private fun setRouteLoaded() {
+        prefs()
+                .edit()
+                .putBoolean(routeLoadedPref, true)
+                .remove(waypointsInProgressPref)
+                .apply()
+    }
+    private fun clearRouteLoaded() {
```

Comment:
I find code generally easier to read with a blank line between consecutive methods.